evm_loader/program/src/account/mod.rs

```diff
@@ -188,6 +193,59 @@ pub fn revision(program_id: &Pubkey, info: &AccountInfo) -> Result<u32> {
     }
 }
 
+// Write heap offset.
+pub fn write_heap_offset(account: &AccountInfo, actual_heap_offset: usize) {
+    let data = account.data.borrow();
+    #[allow(clippy::cast_ptr_alignment)]
+    let offset_ptr = data
+        .as_ptr()
+        .wrapping_add(HEAP_OFFSET_PTR)
+        .cast::<usize>()
+        .cast_mut();
+    unsafe { std::ptr::write_unaligned(offset_ptr, actual_heap_offset) };
+}
+
+// Init heap.
+#[must_use]
+pub fn init_heap(account: &AccountInfo, min_heap_object_offset: usize) -> usize {
+    // Calculate the actual heap object ptr and its offset.
+    let (heap_ptr, heap_object_offset) = {
+        // Locate heap object with offset no less than min_heap_object_offset.
+        let mut heap_object_offset = min_heap_object_offset;
+        let data = account.data.borrow();
+        let mut heap_ptr = data.as_ptr().wrapping_add(heap_object_offset);
+
+        // Calculate alignment and offset the heap pointer.
+        let padding = heap_ptr.align_offset(align_of::<Heap>());
+        heap_ptr = heap_ptr.wrapping_add(padding);
+        assert_eq!(heap_ptr.align_offset(align_of::<Heap>()), 0);
+        heap_object_offset += padding;
+
+        (heap_ptr, heap_object_offset)
+    };
+
+    // Initialize the heap
+    let heap_ptr = heap_ptr.cast_mut();
+    unsafe {
+        // First, zero out underlying bytes of the future heap representation.
+        heap_ptr.write_bytes(0, size_of::<Heap>());
```

Comment:
Is it worth using https://docs.rs/zeroize/latest/zeroize/ for something like this (and in `delete()`)

Reply:
thanks for suggestion! I'm a bit hesitant to pull up additional library just for this one case. Are there other cases where we zero memory?

Reply:
All good, feel free to ignore. It was only a suggestion and something to keep in mind in the future.